app/tests/transaction-form.spec.ts

```diff
@@ -0,0 +1,102 @@
+
+import { resetWallet, setUpElectronApp } from './integration-helpers';
+import { createGlobalFeature } from './features/global.feature';
+import { HomeFeature } from './features/home.feature';
+import { initSoftwareWallet } from './features/onboarding.feature';
+
+const PASSWORD = 'hello9*&^*^*dkfskjdfskljdfsj';
+const SEED_PHRASE =
+  'area kid fat gaze foster eyebrow pen heart draft capable lecture attract daughter news glue saddle exact disorder win olive observe burst option wasp';
+const TX_RECIPIENT = 'STWQ9GP5J69F07RX287ECZ8MRWV48C402HF8CBWB';
+const TX_AMOUNT = Math.random().toFixed(6);
+const TX_MEMO = 'test-memo-field000';
+
+function interceptTransactionBroadcast(page: Page): Promise<Buffer> {
+  return new Promise(resolve => {
+    page.on('request', request => {
+      if (request.url() === 'https://stacks-node-api.testnet.stacks.co/v2/transactions') {
+        const requestBody = request.postDataBuffer();
+        if (requestBody === null) return;
+        resolve(requestBody);
+      }
+    });
+  });
+}
+
+const describeOnlyTestnet = process.env.STX_NETWORK === 'testnet' ? describe : describe.skip;
+
+describeOnlyTestnet('Transaction flow', () => {
+  let app: ElectronApplication;
+  let page: Page;
+
+  beforeAll(async () => {
+    app = await setUpElectronApp();
+    page = await app.firstWindow();
+    await initSoftwareWallet(page)(SEED_PHRASE, PASSWORD);
+  });
+
+  afterAll(async () => {
+    await resetWallet(page);
+    await app.close();
+  });
+
+  test('Transaction form', async done => {
+    const globalFeature = createGlobalFeature(page);
+
+    //
+    // Home steps
+    const homeFeature = new HomeFeature(page);
+    await page.waitForSelector(globalFeature.settingsPageSelector);
+
+    await homeFeature.click('sendStxBtn');
+
+    const addressInput = await page.$(homeFeature.select.sendStxFormAddressInput);
+    await addressInput.type(TX_RECIPIENT);
+
+    const amountInput = await page.$(homeFeature.select.sendStxFormAmountInput);
+    await amountInput.type(TX_AMOUNT);
+
+    const memoInput = await page.$(homeFeature.select.sendStxFormMemoInput);
+    await memoInput.type(TX_MEMO);
+
+    const previewTxBtn = await page.$(homeFeature.select.sendStxFormPreviewBtn);
+    await previewTxBtn.click();
+
+    await delay(1000);
+
+    const memoPreview = await page.$(`text="${TX_MEMO}"`);
+    expect(memoPreview).not.toBeNull();
+
+    await homeFeature.waitFor('sendStxFormSendBtn');
+
+    const stxFormSendBtn = await page.$(homeFeature.select.sendStxFormSendBtn);
+    await stxFormSendBtn.click();
+
+    await homeFeature.waitFor('decryptWalletInput');
+
+    const decryptWalletPasswordInput = await page.$(homeFeature.select.decryptWalletInput);
+    await decryptWalletPasswordInput.type(PASSWORD);
+
+    const broadcastTxBtn = await page.$(homeFeature.select.sendStxFormBroadcastBtn);
+    await broadcastTxBtn.click();
+
+    const requestBody = await interceptTransactionBroadcast(page);
```

Comment:
I'm a bit confused how this works - shouldn't this come before clicking the broadcast tx button? Seems like this could cause a race condition. Perhaps we'd want to have a way of keeping track of previous requests, so then we can query the most recent request.

Reply:
Good observation, though not sure how this would be addressed. Splitting the listener init and promise into two calls? 🤔

Reply:
Yeah basically. Like how [`jest-fetch-mock`](https://www.npmjs.com/package/jest-fetch-mock) works